docs(connectors): Updates connector content and adds auto-restart config (#7909)
strimzi/strimzi-kafka-operator: documentation/assemblies/deploying/assembly-deploy-kafka-connect-with-plugins.adoc

// deploying/assembly-deploy-kafka-connect.adoc

[id='using-kafka-connect-with-plug-ins-{context}']
= Adding connectors

[role="_abstract"]
Kafka Connect uses connectors to integrate with other systems to stream data.
A connector is an instance of a Kafka `Connector` class, which can be one of the following type:

Source connector:: A source connector is a runtime entity that fetches data from an external system and feeds it to Kafka as messages.
Sink connector:: A sink connector is a runtime entity that fetches messages from Kafka topics and feeds them to an external system.

include::../../shared/snip-intro-connectors.adoc[]

Add connector plugins to Kafka Connect in one of the following ways:

* xref:creating-new-image-using-kafka-connect-build-{context}[Configure Kafka Connect to build a new container image with plugins automatically]
* xref:creating-new-image-from-base-{context}[Create a Docker image from the base Kafka Connect image] (manually or using continuous integration)

> After plugins have been added to the container image, you can manage the running of connector instances in the following ways:

Reviewer: ```suggestion
After plugins have been added to the container image, you can start, stop and manage connector instances in the following ways:
```

* xref:proc-deploying-kafkaconnector-{context}[Using Strimzi’s `KafkaConnector` custom resource]
* xref:con-exposing-kafka-connect-api-{context}[Using the Kafka Connect API] 

You can also create new connector instances using these options.

> IMPORTANT: You create the configuration for connectors directly using `KafkaConnector` custom resources or the Kafka Connect REST API.

Reviewer 2: Isn't this essentially the same as the bullet points on lin 24+25?

Reviewer: I agree, I think this line and the one on 27 seem overkill. I suggested a change to line 22 that I think means you could remove lines 27 and 29.

Author: Removed
